src/wallet/coincontrol.h

```diff
@@ -76,8 +77,24 @@ class CCoinControl
         vOutpoints.assign(setSelected.begin(), setSelected.end());
     }
 
+    void AddKnownCoins(const CInputCoin& coin)
+    {
+        knownCoins.insert(std::make_pair(coin.outpoint, coin));
+    }
+
+    boost::optional<CInputCoin> FindKnownCoin(const COutPoint& outpoint) const
```

Comment:
why? We are already using `boost::optional` from https://github.com/bitcoin/bitcoin/pull/10165/files#diff-b2bb174788c7409b671c46ccc86034bdR2088 . This will be implemented in `std::` in C++0x17 as well. I was kind of against it on https://github.com/bitcoin/bitcoin/pull/10165, but it turns out to be very handy.